fdbclient/SystemData.cpp

```diff
@@ -215,6 +215,33 @@ const KeyRangeRef writeConflictRangeKeysRange =
 
 const KeyRef clusterIdKey = LiteralStringRef("\xff/clusterId");
 
+const KeyRef checkpointPrefix = "\xff/checkpoint/"_sr;
+
+const Key checkpointKeyFor(UID checkpointID) {
+	BinaryWriter wr(Unversioned());
+	wr.serializeBytes(checkpointPrefix);
+	wr << checkpointID;
+	return wr.toValue();
+}
+
+const Value checkpointValue(const CheckpointMetaData& checkpoint) {
+	BinaryWriter wr(IncludeVersion());
```

Comment:
Use `ObjectWriter` instead of `BinaryWriter` as the latter makes structures hard to evolve.  `ObjectWriter` will use the same flatbuffers based serialization as the FlowTransport requests/replies.